File: QuartzNetWebConsole/LogEntry.cs
using System;

namespace QuartzNetWebConsole {
    public class LogEntry {
        private readonly DateTimeOffset timestamp;
        private readonly string description;

        public DateTimeOffset Timestamp {
            get { return timestamp; }
        }

        public string Description {
            get { return description; }
        }

        public LogEntry(string description) {
            timestamp = DateTimeOffset.Now;
            this.description = description;
        }
    }
}

File: QuartzNetWebConsole.Views/LogEntry.vb
Public Class LogEntry
    Public ReadOnly Timestamp As DateTimeOffset
    Public ReadOnly Description As String

    Public Sub New(timestamp As DateTimeOffset, description As String)
        Me.Timestamp = timestamp
        Me.Description = description
    End Sub

    Public Sub New(description As String)
        Me.Description = description
        Timestamp = DateTimeOffset.Now
    End Sub
End Class
